public/public.php

```diff
@@ -65,7 +65,7 @@ public function content_filter($content)
             // display inline sources
             $options = $this->get_isc_options();
             if (isset($options['display_type']) && is_array($options['display_type']) && in_array('overlay', $options['display_type'])) {
-                $pattern = '#(\[caption.*align="(.+)"[^\]*]{0,}\])? *(<a [^>]+>)? *(<img .*class=".*(alignleft|alignright|alignnone|aligncenter)??.*wp-image-(\d+)\D*".*src="(.+)".*/?>).*(?(3)(?:</a>)|.*).*(?(1)(?:\[/caption\])|.*)#isU';
+                $pattern = '#(\[caption.*align="(.+)"[^\]*]{0,}\])? *(<a [^>]+>)? *(<img [^>]*class="[^"]*(alignleft|alignright|alignnone|aligncenter)??[^"]*wp-image-(\d+)\D*"[^>]*src="(.+)".*/?>).*(?(3)(?:</a>)|.*).*(?(1)(?:\[/caption\])|.*)#isU';
```

Comment:
Just a quick note-dump:

The regex misses single quoted tags, has uncertain unicode support (`u`-modifier recommended and usage of unicode-classes like `\pL` when really needed) and has some other hard coded byte dependencies like spaces and un-empty `<a>`-tags.

Greediness-settings and matching seems a bit off performance-wise (may want to use `(?:..)` for non-matching parenthesis; use named parenthesis or constant numbering. Naming magic numbers is plain weird to my understanding).